devosoft/Empirical
# Directory reorganization

This pull request addresses issue #378 

Major changes:
- Change .h extensions to .hpp
- Change .cc extensions to .hpp
- source/ is now include/emp/
- tests2/ has been removed
  - relocate tests2/unit_tests.hpp to include/emp/testing/
  - relocate examples of unit_tests.hpp usage to examples directory
- The source/tools/ directory has been reorganized (see issue #378)
- apps/ renamed to demos/
- Add test stubs for (non-web) headers that do not have an associated test file
  - this combats the inflated test coverage number we have (and ensures that our header files at least compile when included...)
- Remove deprecated/broken headers (e.g., constexpr headers, ActiveDataMap, etc)
